### DISCUSSION TITLE
Getting started using SSG without production server

### DISCUSSION THREAD

**Discussion by hornta** [vikejs/vike#1121] [Discussion]
We are running a social platform type of app targeted toward the adult scenery and we would like to have our user profile pages deliver Open Graph meta tags. For this we are looking to this plugin. 

I'm thinking of a setup like this:

The request to a user profile page **always** returns the currently built/cached page. The plugin should most likely start to render this page in the background and save the build output to cache when finished. It should preferably only be done for "old" renders but I want to keep this easy for now and have it start the pre-rendering process on each request.

I'm running my website on Vercel and so I guess the `vercel.json` must be altered so that `/user/:username` requests delivers the cached html file for that particular user. If there is none yet, then it returns the default index.html and the rendering process starts in the background.

I'm not so sure after writing this that it's possible. 

- We do have 10.000+ profiles and so worst case there will be 10.000+ generated HTML files. Is it possible to update a vercel deployment when a user profile pages has finished rendering to upsert that HTML file? 
- When pushing new code that could potentially invalidate the HTML of cached profile pages, it would most likely be good to invalidate the cache. 
- I'm thinking for myself what "invalidation" means. It could mean that all cached/built pages for each user are deleted. Or it could mean that the files themselves are marked as invalidated but then I would have to put this information somewhere, maybe utilizing Vercel KV

Pre-rendering the pages upon build is not feasible because new code changes are deployed once a week, if not less often and user profiles changes much more drastically.

**Comment by brillout** [comment]
Have a look at [Vercel's ISR](https://vercel.com/docs/incremental-static-regeneration) and [vite-plugin-vercel](https://github.com/magne4000/vite-plugin-vercel) by @magne4000 which has an integration with vite-plugin-ssr.

**Reply by brillout** [reply]
It seems like ISR is what you're looking for, but feel free to let us know if you need something (else) from vite-plugin-ssr.
